crates/control/src/time_to_reach_kick_position.rs

```diff
@@ -16,67 +14,58 @@ pub struct TimeToReachKickPosition {}
     stand_up_front_estimated_remaining_duration:
-        Input<Option<Duration>, "stand_up_front_estimated_remaining_duration?">,
+        CyclerState<Duration, "stand_up_front_estimated_remaining_duration">,
 }
 
 #[context]
 pub struct CreationContext {}
 
 #[context]
-pub struct MainOutputs {}
+pub struct MainOutputs {
+    pub time_to_reach_kick_position: MainOutput<Duration>,
+}
 
 impl TimeToReachKickPosition {
     pub fn new(_: CreationContext) -> Result<Self> {
         Ok(Self {})
     }
 
-    pub fn cycle(&mut self, mut context: CycleContext) -> Result<MainOutputs> {
-        let walk_time = context
-            .dribble_path
-            .as_ref()
-            .map(|path| {
-                path.iter()
-                    .map(|segment: &PathSegment| {
-                        let length = segment.length();
-                        match segment {
-                            PathSegment::LineSegment(_) => {
-                                length / context.configuration.path_planning.line_walking_speed
-                            }
-                            PathSegment::Arc(_) => {
-                                length / context.configuration.path_planning.arc_walking_speed
-                            }
+    pub fn cycle(&mut self, context: CycleContext) -> Result<MainOutputs> {
+        let Some((orientation_mode, dribble_path)) = context.dribble_path_plan else {
+            return Ok(MainOutputs {
+                time_to_reach_kick_position: Duration::MAX.into(),
+            });
+        };
+        let walk_time = Duration::from_secs_f32(
+            dribble_path
+                .iter()
+                .map(|segment: &PathSegment| {
+                    let length = segment.length();
+                    match segment {
+                        PathSegment::LineSegment(_) => {
+                            length / context.configuration.path_planning.line_walking_speed
                         }
-                    })
-                    .sum()
-            })
-            .map(Duration::from_secs_f32);
-        let turning_angle = match context.motion_command {
-            MotionCommand::Walk {
-                orientation_mode: OrientationMode::Override(orientation),
-                ..
-            } => Some(orientation.angle().abs()),
-            _ => {
-                let turning_angle_towards_path = match context.dribble_path {
-                    Some(path) => match path.first() {
-                        Some(PathSegment::LineSegment(line_segment)) => {
-                            Some(line_segment.1.coords().angle(&Vector2::x_axis()).abs())
+                        PathSegment::Arc(_) => {
+                            length / context.configuration.path_planning.arc_walking_speed
                         }
-                        _ => None,
-                    },
+                    }
+                })
+                .sum(),
+        );
```

Comment:
Might be a little picky, but try to reduce some levels of indentation ;)

e.g.
```rust
        let secs = dribble_path
            .iter()
            .map(|segment: &PathSegment| {
                let length = segment.length();
                match segment {
                    PathSegment::LineSegment(_) => {
                        length / context.configuration.path_planning.line_walking_speed
                    }
                    PathSegment::Arc(_) => {
                        length / context.configuration.path_planning.arc_walking_speed
                    }
                }
            })
            .sum();
        let walk_time = Duration::from_secs_f32(secs);
```

Reply:
Done, also cleaned up the turn duration code below.